store-implementation/map-store/src/main/java/uk/gov/gchq/gaffer/mapstore/impl/DeleteElementsHandler.java

```diff
@@ -0,0 +1,167 @@
+      // Delete all elements directly
+      deleteBatch(mapImpl, schema, elements);
+    } else {
+      LOGGER.info("Deleting elements in batches, batch size = {}", bufferSize);
+      int count = 0;
+      final List<Element> batch = new ArrayList<>(bufferSize);
+      for (final Element element : elements) {
+        if (null != element) {
+          batch.add(mapImpl.cloneElement(element, schema));
+          count++;
+          if (count >= bufferSize) {
+            deleteBatch(mapImpl, schema, AggregatorUtil.ingestAggregate(batch, schema));
+            batch.clear();
+            count = 0;
+          }
+        }
+      }
+
+      if (count > 0) {
+        deleteBatch(mapImpl, schema, AggregatorUtil.ingestAggregate(batch, schema));
+      }
+    }
+  }
+
+  private void deleteBatch(final MapImpl mapImpl, final Schema schema, final Iterable<? extends Element> elements) {
+    for (final Element element : elements) {
+      if (null != element) {
+        final Element elementForIndexing = deleteElement(element, schema, mapImpl);
+
+        // Update entityIdToElements and edgeIdToElements if index required
+        if (mapImpl.isMaintainIndex()) {
+          updateElementIndex(elementForIndexing, mapImpl);
+        }
+      }
+    }
+  }
+
+  private Element deleteElement(final Element element, final Schema schema, final MapImpl mapImpl) {
+    final Element elementForIndexing;
+    if (mapImpl.isAggregationEnabled(element)) {
+      elementForIndexing = deleteAggElement(element, mapImpl);
+    } else {
+      elementForIndexing = deleteNonAggElement(element, schema, mapImpl);
+    }
+    return elementForIndexing;
+  }
+
+  private Element deleteAggElement(final Element element, final MapImpl mapImpl) {
+    final String group = element.getGroup();
+    final Element elementWithGroupByProperties = element.emptyClone();
+    final GroupedProperties properties = new GroupedProperties(element.getGroup());
+    if (null != mapImpl.getGroupByProperties(group)) {
+      for (final String propertyName : mapImpl.getGroupByProperties(group)) {
+        elementWithGroupByProperties.putProperty(propertyName, element.getProperty(propertyName));
+      }
+    }
+    if (null != mapImpl.getNonGroupByProperties(group)) {
+      for (final String propertyName : mapImpl.getNonGroupByProperties(group)) {
+        properties.put(propertyName, element.getProperty(propertyName));
+      }
+    }
+    mapImpl.deleteAggElement(elementWithGroupByProperties);
+
+    return elementWithGroupByProperties;
+  }
+
+  private Element deleteNonAggElement(final Element element, final Schema schema, final MapImpl mapImpl) {
+    final Element elementClone = element.emptyClone();
+
+    // Copy properties that exist in the schema
+    final SchemaElementDefinition elementDef = schema.getElement(element.getGroup());
+    for (final String property : elementDef.getProperties()) {
+      elementClone.putProperty(property, element.getProperty(property));
```

Comment:
Possibly more just styling but simple loops like this might be more readable as lambdas  e.g. this could be
```java
schema.getElement(element.getGroup())
    .getProperties()
    .forEach(p -> elementClone.putProperty(p, element.getProperty(p)));
```